bigquery/google/cloud/bigquery/client.py

```diff
@@ -1529,6 +1530,14 @@ def load_table_from_dataframe(
         os.close(tmpfd)
 
         try:
+            if job_config.schema:
+                for field in job_config.schema:
+                    if field.field_type in _STRUCT_TYPES:
+                        raise ValueError(
+                            "Pyarrow does not support serializing dataframes with "
```

Comment:
I think we can remove the mention of `pyarrow`. It's an implementation detail, and we're going to want to explicitly add support for struct types some time after `pyarrow` adds support.

Instead, let's say `Uploading dataframes with struct (record) column types is not supported. See: https://github.com/googleapis/google-cloud-python/issues/8191`